firestore-translate-text/functions/src/index.ts

```diff
@@ -173,13 +194,58 @@ const translateDocument = async (
       {}
     );
 
-    await updateTranslations(snapshot, translationsMap);
+    return updateTranslations(snapshot, translationsMap);
   } catch (err) {
     logs.translateInputToAllLanguagesError(input, err);
     throw err;
   }
 };
 
+const translateMultiple = async (
+  input: object,
+  snapshot: admin.firestore.DocumentSnapshot
+): Promise<void> => {
+  let translations = {};
+  let promises = [];
+
+  Object.entries(input).forEach(([input, value]) => {
+    config.languages.forEach((language) => {
+      promises.push(
+        () =>
+          new Promise(async (resolve) => {
+            logs.translateInputStringToAllLanguages(value, config.languages);
+
+            const output = await translateString(value, language);
+
+            if (!translations[input]) translations[input] = {};
+            translations[input][language] = output;
+
+            return resolve();
+          })
+      );
+    });
+  });
+
+  for (const fn of promises) {
+    if (fn) await fn();
+  }
+
+  return updateTranslations(snapshot, translations);
+};
+
+const translateDocument = async (
+  snapshot: admin.firestore.DocumentSnapshot
+): Promise<void> => {
+  const input: string = extractInput(snapshot);
+
+  if (typeof input === "object") {
+    await translateMultiple(input, snapshot);
+    return Promise.resolve();
```

Comment:
Can just `return` here, since it's already a promise, rather than creating a new instance.